src/test/java/org/assertj/core/api/recursive/comparison/DualValue_iterableValues_Test.java

```diff
@@ -131,8 +142,9 @@ void isExpectedFieldAnIterable_should_return_false_when_expected_is_not_an_itera
   }
 
   static Stream<Object> nonIterables() {
-    // even though Path is an iterable, it must not be considered as such
-    return Stream.of(123, "abc", array("a", "b"), Paths.get("/tmp"));
+    // even though Path and ValueNode are iterables, they must not be considered as such
+    return Stream.of(123, "abc", array("a", "b"), Paths.get("/tmp"), new TextNode("foo"), new IntNode(42), BooleanNode.TRUE,
+                     new POJONode(new Object()), MissingNode.getInstance(), NullNode.getInstance(), new BinaryNode(new byte[0]));
```

Comment:
Instantiating directly Jackson types together with other iterables/non-iterables seems fragile to me.

I would rather try to replicate what end users would do, i.e., `new ObjectMapper().readTree("...")`, and compose different test cases as input of `readTree` in a separate parameterized test(s).

However, I'm not sure yet how it would look like. I can try to play with it and come up with a proposal, but don't wait for me in case you have already an idea 🙂

Reply:
Maybe something in this spirit?

```java
  @ParameterizedTest
  @MethodSource("nonIterableJsonNodes")
  void isExpectedFieldAnIterable_should_return_false_when_expected_is_a_non_iterable_json_node(JsonNode expected) {
    // GIVEN
    DualValue dualValue = new DualValue(PATH, "", expected.findValue("value"));
    // WHEN
    boolean isExpectedFieldAnIterable = dualValue.isExpectedFieldAnIterable();
    // THEN
    assertThat(isExpectedFieldAnIterable).isFalse();
  }

  static Stream<JsonNode> nonIterableJsonNodes() {
    // even though ValueNode is an iterable, it must not be considered as such
    return Stream.of("{\"value\": \"foo\"}", "{\"value\": 42}", "{\"value\": true}")
                 .map(json -> toJsonNode(json));
  }

  private static JsonNode toJsonNode(String value) {
    ObjectMapper objectMapper = new ObjectMapper();
    try {
      return objectMapper.readTree(value);
    } catch (JsonProcessingException e) {
      throw new RuntimeException(e);
    }
  }
```
and another test for `"{\"value\": []}"` and `"{\"value\": {}}"`